pkg/target/target.go

```diff
@@ -57,22 +62,24 @@ type Target interface {
 }
 
 type targetImpl struct {
-	Garden  string `yaml:"garden,omitempty" json:"garden,omitempty"`
-	Project string `yaml:"project,omitempty" json:"project,omitempty"`
-	Seed    string `yaml:"seed,omitempty" json:"seed,omitempty"`
-	Shoot   string `yaml:"shoot,omitempty" json:"shoot,omitempty"`
+	Garden       string `yaml:"garden,omitempty" json:"garden,omitempty"`
+	Project      string `yaml:"project,omitempty" json:"project,omitempty"`
+	Seed         string `yaml:"seed,omitempty" json:"seed,omitempty"`
+	Shoot        string `yaml:"shoot,omitempty" json:"shoot,omitempty"`
+	ControlPlane bool   `yaml:"controlPlane,omitempty" json:"controlPlane,omitempty"`
 }
 
 var _ Target = &targetImpl{}
 
 // NewTarget returns a new target. This function does not perform any validation,
 // so the returned target can be invalid.
-func NewTarget(gardenName, projectName, seedName, shootName string) Target {
+func NewTarget(gardenName, projectName, seedName, shootName string, shootControlPlane bool) Target {
```

Comment:
Do not change the signature of the `NewTarget` function which would result in the `ControlPlane` property to be `false`. Add a new function called `NewControlPlaneTarget` with the same parameters but `ControlPlane` property set to true or use the WithControlPlane method where needed.

Reply:
ok